### PR TITLE
adds basic implementation of 'qx add'

### PR DESCRIPTION
**Pull request by cboulanger**
This is an implementation of `qx add class` as outlined in #82. Contrib support hasn't been implemented yet. 

(updated to reflect current code)

```
/**
 * Add a new class file to the current project, based on a template.
 *
 * Syntax: `qx add class <classname> [--type=template_name] [--extend=extended_class] [--import] [--from-contrib=contrib_name]`
 * If omitted, `--type` defaults to "default". The path to the template file will be calculated as follows:
 * 1. transform template_name to ${template_name}.tmpl.js
 * 2. check if this file exists in the "templates/class" folder 
 *    a. of a contrib, if the --from-contrib paramater wass passed (not implemented)
 *    b. otherwise, of the current project
 *    c. finally, of the CLI library
 *
 * If the --import flag is set, copy the template to the templates/class folder of the current project,
 * so it can be customized and used instead of the one shipped with the CLI.
 * 
 * If you place a file named `header.js` in the root of your project, this header will be used verbatim
 * as the ${header} template variable instead of the generic header template, which is populated with 
 * information from Manifest.json
 * 
 * (Contrib support is not yet implemented)
 * 
 */
```

### CODE REVIEW

File: templates/class/default.tmpl.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  {
+    /** The foo static property of the class object */
+    FOO : "bar"
+  },  
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
```
Comment (derrell) [inline templates/class/default.tmpl.js]: Do you guys actually retain these comment blocks for MEMBER, EVENTS, etc? The first thing I do, if copying something from a framework class, is get rid of that cruft. It's marked as `members :` immediately below, so really doesn't add any benefit at all, as far as I can see...

Comment (cajus) [reply templates/class/default.tmpl.js]: 👍

Comment (cboulanger) [reply templates/class/default.tmpl.js]: Well, I think the 1&1 guys added the comments for better readability of long classes (you can jump much faster between code blocks with these visual markers), and it still might some value for beginners. But I certainly don't insist on keeping them.

Comment (cboulanger) [reply templates/class/default.tmpl.js]: I just asked the question in the core-team chat, but maybe that would also be a question for the community... Is that a show stopper for you or could we go with the verbose templates for the moment? We can still change them later and in any case I don't think you will use the standard templates anyways, but create your own ones if you use them at all.

**Comment by cboulanger** [comment]
To recap, that's the current state of the implementation:

```
/**
 * Add a new class file to the current project, based on a template.
 *
 * Syntax: `qx add class <classname> [--type=template_name] [--extend=extended_class] [--import] [--from-contrib=contrib_name]`
 * If omitted, `--type` defaults to "default". The path to the template file will be calculated as follows:
 * 1. transform template_name to ${template_name}.tmpl.js
 * 2. check if this file exists in the "templates/class" folder 
 *    a. of a contrib, if the --from-contrib paramater wass passed (not implemented)
 *    b. otherwise, of the current project
 *    c. finally, of the CLI library
 *
 * If the --import flag is set, copy the template to the templates/class folder of the current project,
 * so it can be customized and used instead of the one shipped with the CLI.
 * 
 * If you place a file named `header.js` in the root of your project, this header will be used verbatim
 * as the ${header} template variable instead of the generic header template, which is populated with 
 * information from Manifest.json
 * 
 * (Contrib support is not yet implemented)
 * 
 */
```

**Comment by cboulanger** [comment]
Ok, I replaced the big fat comments with smaller ones containing explanations. As I said before, the default templates will be mainly used by beginners and in tutorials, so I think those are useful. The templates should show as much functionality of the class system as possible, but still be usable. If you have more suggestions how to improve them, I would ask you to do that in separate PRs, so that we get the main feature into qooxdoo-cli now.

**Review by hkollmann — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Comment by johnspackman** [comment]
5 approvals, merging. thank you @cboulanger !

**Pull request merged**